lsp-mode.el

```diff
@@ -7081,22 +7109,24 @@ When prefix UPDATE? is t force installation even if the server is present."
 
 (defun lsp-async-start-process (callback error-callback &rest command)
   "Start async process COMMAND with CALLBACK and ERROR-CALLBACK."
-  (make-process
-   :name (cl-first command)
-   :command command
-   :sentinel (lambda (proc _)
-               (when (eq 'exit (process-status proc))
-                 (if (zerop (process-exit-status proc))
-                     (condition-case err
-                         (funcall callback)
-                       (error
-                        (funcall error-callback (error-message-string err))))
-                   (display-buffer " *lsp-install*")
-                   (funcall error-callback
-                            (format "Async process '%s' failed with exit code %d"
-                                    (process-name proc) (process-exit-status proc))))))
-   :buffer " *lsp-install*"
-   :noquery t))
+  (let ((name (cl-first command)))
+    (with-current-buffer (compilation-start (combine-and-quote-strings command) t
+                                            (lambda (&rest _)
+                                              (generate-new-buffer-name (format "*lsp-install: %s*" name))))
+      (setq-local lsp--installation-buffer t)
```

Comment:
This is already local buffer var, `setq is enough?

Reply:
Imho `setq-local` is clearer in such cases: the `defvar-local` tells the reader that the variable will always be local, and the `setq-local` here also does. This way, one needn't jump to `lsp--installation-buffer` first to check if it is local.

Reply:
Well, I just want to write code without worrying about variable is buffer local or not, so using ubiquitous `setq` everywhere.
The `setq-local` is just a macro that expanded to `(set (make-local-variable xyz) <value>)` so we spends cycles to make variable local everytime this macro is called.
Declaring variable as local once also make it easier if we decide to change the variable back to global, just changing from `defvar-local` -> `defvar` and it will be fine.

Reply:
The performance overhead is insanely small: doing `make-local-variable` a million times takes less than a second, and in this case, the variable really has to be buffer local for it to make any sense. With that in mind, what do you think? Should I prefer `setq`? There are arguments for both, since the variable's name also implies that it would be local.

Benchmark:
```emacs-lisp
(benchmark-run 1000000 (make-local-variable 'abc))
(0.070333... 0 0.0)
```

Reply:
If you are interested what is the proper way, ask in emacs-devel what is the convention to use defvar-local variables. Except for the naming we tend to follow their conventions.